site/app/models/User.php

```diff
@@ -188,6 +191,15 @@ public function __construct(Core $core, $details = []) {
             $this->setGradingRegistrationSections($details['grading_registration_sections']);
         }
 
+        if (isset($details['display_image_state'])) {
+            try {
+                $this->display_image = new DisplayImage($core, $details['user_id'], $details['display_image_state']);
+            }
+            catch (\Exception $exception) {
+                $this->display_image = null;
+            }
```

Comment:
Can we just store `$details['display_image_state']` to `$this->display_image` and then make a concrete `getDisplayImage()` that does the `DisplayImage` construction. If we find we're calling `getDisplayImage` multiple times on the same User model, then we can cache the result in that function, but not sure we are.

We should not waste the cycles on constructing the model unless we actually are going to call this function, which I do not think we do in a large number of cases.

Reply:
I'm going to shelve this until a later PR.  See #5587.